docs/gallery/howto/autogen/parallel.py

```diff
@@ -1,148 +1,280 @@
+    add10_1_task.ctime.time(),
+    "finished:",
+    add10_1_task.mtime.time(),
+)
 
+# %%
+# We can see that the time is less than 6 seconds which means that the two additions
+# were performed in parallel
 
 # %%
-# Now, let's create a `WorkGraph` to use the new task:
-#
+# We can also look at the total time and see the overhead costs.
+print(
+    "Time for running parallelized graph builder",
+    parallel_add_task.mtime - parallel_add_task.ctime,
+)
 
-from aiida_workgraph import WorkGraph
-from aiida.orm import Int, List
+# %%
+# Increasing number of daemon workers
+# -----------------------------------
+# Since each daemon worker can only manage one WorkGraph (handling the results)
+# at a time, one can experience slow downs when running many jobs that can be
+# run in parallel. The optimal number of workers depends highly on the jobs
+# that are run.
 
-X = {"a": Int(1), "b": Int(2), "c": Int(3)}
-y = Int(2)
-z = Int(3)
-wg = WorkGraph("parallel_tasks")
-multiply_parallel_gather1 = wg.add_task(multiply_parallel_gather, X=X, y=y)
-sum1 = wg.add_task(sum, name="sum1")
-# wg.add_link(add1.outputs[0], multiply_parallel_gather1.inputs["uuids"])
-wg.add_link(multiply_parallel_gather1.outputs[0], sum1.inputs[0])
+from aiida.engine.daemon.client import get_daemon_client
 
-wg.submit(wait=True)
+client = get_daemon_client()
+print(f"Number of current daemon workers {client.get_numprocesses()['numprocesses']}")
 
 # %%
-# Get the result of the tasks:
-#
+# We rerun the last graph builder with 2 damon workers
 
-print("State of WorkGraph:   {}".format(wg.state))
-print("Result of task add1: {}".format(wg.tasks["sum1"].outputs["result"].value))
+client.increase_workers(1)
+print(f"Number of current daemon workers {client.get_numprocesses()['numprocesses']}")
+wg = WorkGraph("wg_daemon_worker_2")
+parallel_add_task_2 = wg.add_task(parallel_add, name="parallel_add", nb_iterations=2)
+wg.to_html()
 
+# %%
+wg.submit(wait=True)
+print(
+    "Time for running parallelized graph builder with 2 daemons",
+    parallel_add_task_2.mtime - parallel_add_task_2.ctime,
+)
 
 # %%
-# Generate node graph from the AiiDA process:
-#
+# The overhead time has shortens a bit as the handling of the CalcJobs and
+# WorkGraphs could be parallelized. One can increase the number of iterations
+# to see a more significant difference.
 
-from aiida_workgraph.utils import generate_node_graph
 
-generate_node_graph(wg.pk)
+# %%
+# Reset back to one worker
+client.decrease_workers(1)
 
 # %%
-# You can see that the outputs of `multiply_parallel_gather` workgraph is linked to the input of the `sum` task.
+# Maximum number of active WorkGraphs
+# -----------------------------------
+# Be aware that for the moment AiiDA can only run 200 WorkGraphs at the same time.
+# To increase that limit one can set this variable to a higher value.
+#
+#     .. code-block:: bash
 #
+#        verdi config set daemon.worker_process_slots 200
+#        verdi daemon restart
+
```

Comment:
It would be good to add a link to the [performance page](https://aiida.readthedocs.io/projects/aiida-core/en/latest/howto/installation.html#tuning-performance).